pandas/io/excel/_openpyxl.py

```diff
@@ -586,9 +586,12 @@ def _convert_cell(self, cell) -> Scalar:
         elif cell.data_type == TYPE_ERROR:
             return np.nan
         elif cell.data_type == TYPE_NUMERIC:
-            val = int(cell.value)
-            if val == cell.value:
-                return val
+            # If we try to convert a large float to an integer, weird issues arise because of precision limitation of
+            # floating point numbers
+            if abs(cell.value) < 1e22:
```

Comment:
Thanks but these constants like 1e22 are not very robust to platform difference and can hide bugs. Unless there's a openpyxl flag or package constant that can be used, I recommend just documenting this limitation.

Reply:
Since we know Python represents floats as 64 bit doubles, we are guaranteed that all doubles have 16 decimal digits of precision (according to https://en.wikipedia.org/wiki/Double-precision_floating-point_format). So I'm thinking we can replace 1e22 with 1e16? `1e22` was just an empirically determined value, but `1e16` is based off the maximum precision of a double number.

On a practical note, numbers larger than `1e16` are probably better represented with floats than integers. If an Excel cell contains numbers that large, the meaning is better captured as a float than as an integer. 

For context, this `int(cell.value)` check was done because of https://github.com/pandas-dev/pandas/issues/46988.